test/FastTests/Server/Documents/Revisions/RevisionsTests.cs

```diff
@@ -2146,6 +2147,120 @@ public async Task CanGetRevisionsCountForAsync(Options options)
             }
         }
 
+        [RavenTheory(RavenTestCategory.ClientApi | RavenTestCategory.Revisions)]
+        [RavenData(DatabaseMode = RavenDatabaseMode.All)]
+        public async Task GetRevisionsByIdShouldGetNotModifiedStatus(Options options)
+        {
+            var company = new Company { Name = "Company Name" };
+            using (var store = GetDocumentStore(options))
+            {
+                await RevisionsHelper.SetupRevisionsAsync(store);
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(company);
+                    await session.SaveChangesAsync();
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var company2 = await session.LoadAsync<Company>(company.Id);
+                    company2.Name = "Hibernating Rhinos";
+                    await session.SaveChangesAsync();
+                }
+
+                var serverNode = new ServerNode
+                {
+                    Database = store.Database,
+                    Url = store.Urls.First()
+                };
+
+                using (var re = store.GetRequestExecutor())
+                using (re.ContextPool.AllocateOperationContext(out var ctx))
+                {
+                    var revisionsCommand = new GetRevisionsCommand(company.Id, before: DateTime.MaxValue);
+                    var request = revisionsCommand.CreateRequest(ctx, serverNode, out var url);
+                    request.RequestUri = new UriBuilder(url).Uri;
+
+                    var response = await re.HttpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, CancellationToken.None);
+
+                    Assert.True(response.StatusCode != HttpStatusCode.NotModified);
+
+                    var etagFromResponse = response.Headers.ETag?.Tag;
+
+                    request = revisionsCommand.CreateRequest(ctx, serverNode, out url);
+                    request.RequestUri = new UriBuilder(url).Uri;
+                    request.Headers.IfNoneMatch.ParseAdd(etagFromResponse);
+
+                    response = await re.HttpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, CancellationToken.None);
+                    
+                    Assert.True(response.StatusCode == HttpStatusCode.NotModified);
+                }
```

Comment:
Would it be simpler to use the `GetRevisionOperation` and hook to the `OnSucceedRequest` ?